### ISSUE TITLE
Wikilinks not parsed on first pass

### ISSUE THREAD

**Issue by kjschiroo** [earwig/mwparserfromhell#160]
For revision [695594020.txt](https://github.com/earwig/mwparserfromhell/files/423734/695594020.txt) some links are not being parsed on the first pass.

To demonstrate this:

```
import mwparserfromhell as mwp

text = open("695594020.txt").read()
wcode = mwp.parse(text)
for node in wcode.nodes:
    if "04:42, 12 December 2015 (UTC)" in node:
        print(node)
        print(type(node))
        print("**************************************")
        break
wcode = mwp.parse(str(node))
for node in wcode.nodes:
    print(node)
    print(type(node))
```

Gives this as output:

```
Hello TeaHouse !

I wondering or curious about why my article Data Excellence submission was declined.
I forward you the message of a reviewer: This submission seems to be a test edit and not an article worthy of an encyclopedia. Please use the sandbox for any editing tests, but do not submit for review until you have an article that you want reviewed for inclusion in Wikipedia. Thank you.

I don't understand really what can i do ?
Can you help me ? [[User:Yanniyolo|Yanniyolo]] ([[User talk:Yanniyolo|talk]]) 04:42, 12 December 2015 (UTC)


<class 'mwparserfromhell.nodes.text.Text'>
**************************************

Hello TeaHouse !

I wondering or curious about why my article Data Excellence submission was declined.
I forward you the message of a reviewer: This submission seems to be a test edit and not an article worthy of an encyclopedia. Please use the sandbox for any editing tests, but do not submit for review until you have an article that you want reviewed for inclusion in Wikipedia. Thank you.

I don't understand really what can i do ?
Can you help me ? 
<class 'mwparserfromhell.nodes.text.Text'>
[[User:Yanniyolo|Yanniyolo]]
<class 'mwparserfromhell.nodes.wikilink.Wikilink'>
 (
<class 'mwparserfromhell.nodes.text.Text'>
[[User talk:Yanniyolo|talk]]
<class 'mwparserfromhell.nodes.wikilink.Wikilink'>
) 04:42, 12 December 2015 (UTC)


<class 'mwparserfromhell.nodes.text.Text'>

```

**Comment by lahwaacz** [comment]
That's #40. The revision contains couple of unclosed `<code>` tags (e.g. `<code><nowiki><ref>source_information</ref></nowiki><code>`) and mwparserfromhell chokes on this.

**Comment by earwig** [comment]
It's always #40. When am I ever gonna get around to fixing that? _sighs_

**Comment by earwig** [comment]
I think we can consider this fixed by #42, since the example now produces the links as expected.
